apps/transmitmedia.cpp

```diff
@@ -650,11 +691,11 @@ int SrtSource::Read(size_t chunk, bytevector& data, ostream &out_stats)
         srt_bstats(m_sock, &perf, need_stats_report && !transmit_total_stats);
         if (need_bw_report)
         {
-            PrintSrtBandwidth(perf.mbpsBandwidth);
+            cerr << stats_writer->WriteBandwidth(perf.mbpsBandwidth) << std::flush;
         }
         if (need_stats_report)
         {
-            PrintSrtStats(m_sock, perf, out_stats);
+            out_stats << stats_writer->WriteStats(m_sock, perf) << std::flush;
```

Comment:
`need_stats_report` and `need_bw_report` should include `stats_writer != nullptr` check.

Reply:
Good catch! Fixed.